Phantom deeplinks

phoenixVS opened the pull request:
# Breaking Changes

N/A

# Changes

- feat:
- fix:
- chore:

# Associated Issues

closes #...

svenvoskamp commented on packages/scaffold/src/views/w3m-connect-view/index.ts:
```diff
@@ -145,7 +145,7 @@ export class W3mConnectView extends LitElement {
   private announcedTemplate() {
     return this.connectors.map(connector => {
-      if (connector.type !== 'ANNOUNCED' || CoreHelperUtil.isMobile()) {
+      if (connector.type !== 'ANNOUNCED') {
```
We removed this check in the hotfix this morning, this was indeed causing issues that announced wallets didn't show up

svenvoskamp reviewed (changes requested):
Could you please fix the merge conflicts? 

We only need to the changes from `src/utils/defaultConfig.ts` and `solana/src/client.ts`

enesozturk commented:
> Could you please fix the merge conflicts?
> 
> We only need to the changes from `src/utils/defaultConfig.ts` and `solana/src/client.ts`

@phoenixVS

enesozturk commented on apps/laboratory/src/components/Solana/SolanaSignTransactionTest.tsx:
```diff
@@ -13,7 +13,7 @@ import { useWeb3ModalAccount, useWeb3ModalProvider } from '@web3modal/solana/rea
 import { useChakraToast } from '../Toast'
 
-const PHANTOM_DEVNET_ADDRESS = 'EmT8r4E8ZjoQgt8sXGbaWBRMKfUXsVT1wonoSnJZ4nBn'
+const PHANTOM_DEVNET_ADDRESS = '8vCyX7oB6Pc3pbWMGYYZF5pbSnAdQ7Gyr32JqxqCy8ZR'
```
Where do we get these addresses? @phoenixVS

phoenixVS replied:
No that is just a wallet for an example of a transaction
Can be anything. I changed to one I have access to so I can more easier debug trx workflow

phoenixVS commented:
> Could you please fix the merge conflicts?
> 
> We only need to the changes from `src/utils/defaultConfig.ts` and `solana/src/client.ts`

Fixed, only added a few to laboratory test small change
It's just a test wallet address, so I've changed it to one I have access to and debug transactions easier

phoenixVS commented on packages/solana/src/client.ts:
```diff
@@ -324,6 +347,8 @@ export class Web3Modal extends Web3ModalScaffold {
   private syncConnectors() {
     const w3mConnectors: Connector[] = []
 
+    syncInjectedWallets(w3mConnectors, this.walletAdapters)
```
No, same if on bottom, I think I would revert this change

phoenixVS commented on packages/solana/src/client.ts:
```diff
@@ -255,9 +255,32 @@ export class Web3Modal extends Web3ModalScaffold {
+        }
+      }
+
+      setTimeout(() => {
```
The problem is Brave, sometime doesn't add [solflare|trust|backpack] objects to Window for some delay
Haven't found any sources, but left this workaround for now

I will left more detailed explanation in comment

enesozturk commented:
> I will left more detailed explanation in comment

@phoenixVS Please add some sources
